ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/home/view/HomeView.kt

```diff
@@ -189,33 +209,160 @@ private fun HomeMainContent(
+
         History(transactionHistory)
     }
 }
 
 @Composable
-private fun Status(walletSnapshot: WalletSnapshot) {
-    Column(Modifier.fillMaxWidth()) {
-        Header(text = walletSnapshot.totalBalance().toZecString())
-        Body(
-            text = stringResource(
-                id = R.string.home_status_shielding_format,
-                walletSnapshot.saplingBalance.total.toZecString()
+@Suppress("LongMethod", "MagicNumber")
+private fun Status(walletSnapshot: WalletSnapshot, updateAvailable: Boolean) {
+    // TODO this will go away before PR merged
+    Twig.info { "WALLET: $walletSnapshot" }
+
+    val configuration = LocalConfiguration.current
+    val contentSizeRatioRatio = if (configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        0.45f
+    } else {
+        0.9f
+    }
+
+    // parts sizes
+    val progressCircleStroke = 12.dp
+    val progressCirclePadding = progressCircleStroke + 6.dp
+    val contentPadding = progressCircleStroke + progressCirclePadding + 10.dp
+
+    // parts values
+    var progressCirclePercentage = 0
+    val zecAmountText = walletSnapshot.totalBalance().toZecString()
+    var usdAmountText = walletSnapshot.spendableBalance().toUsdString()
+    var statusText = ""
+
+    // Note: these reactions on the STATUS need to be enhanced, we provide just an elementary reactions for now.
+    when (walletSnapshot.status) {
+        Synchronizer.Status.PREPARING,
+        Synchronizer.Status.DOWNLOADING,
+        Synchronizer.Status.VALIDATING -> {
+            progressCirclePercentage = walletSnapshot.progress
+            usdAmountText = stringResource(
+                R.string.home_status_syncing_amount_suffix,
+                walletSnapshot.spendableBalance().toUsdString()
+            )
+            statusText = stringResource(R.string.home_status_syncing_format, walletSnapshot.progress)
+        }
+        Synchronizer.Status.SCANNING -> {
+            // SDK provides us only one progress, which keeps on 100 in the scanning state
+            progressCirclePercentage = 100
+            statusText = stringResource(R.string.home_status_syncing_catchup)
+        }
+        Synchronizer.Status.SYNCED,
+        Synchronizer.Status.ENHANCING -> {
+            statusText = if (updateAvailable) {
+                stringResource(R.string.home_status_update)
+            } else {
+                stringResource(R.string.home_status_up_to_date)
+            }
+        }
+        Synchronizer.Status.DISCONNECTED -> {
+            statusText = stringResource(
+                R.string.home_status_error,
+                stringResource(R.string.home_status_error_connection)
             )
+        }
+        Synchronizer.Status.STOPPED -> {
+            statusText = stringResource(R.string.home_status_stopped)
+        }
+    }
+
+    // more detailed error message
+    if (walletSnapshot.hasSynchronizerError) {
```

Comment:
Could we do this?
```
walletSnapshot.synchronizerError?.let {

}
```

I think that would eliminate the `!!` below.

Reply:
Yep, we don't need that.